tools/sof-ipc-timer.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+'''The sof-ipc-timer collects module initialization and configuration
+timings from 'journalctl -k -o short-precise' output if SOF IPC debug
+is enabled.
+
+'''
+
+import re
+import pathlib
+import argparse
+from datetime import datetime
+
+class Component:
+    '''SOF audio component storage class'''
+    ppln_id: str
+    comp_id: int
+    wname: str
+
+    def __init__(self, ppln_id, comp_id, wname):
+        self.ppln_id = ppln_id
+        self.comp_id = comp_id
+        self.wname = wname
+
+    def __str__(self) -> str:
+        return f'{self.ppln_id}-{self.comp_id:#08x}'
+
+def read_comp_data(f):
+    '''Process the dmesg to get the component ID to component name mapping,
```

Comment:
```suggestion
    '''Process the kernel logs to get the component ID to component name mapping,
```
`dmesg` wraps around. `journalctl -k` does not.